src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TextBox.skia.cs

```diff
@@ -4607,6 +4608,105 @@ public async Task When_Rearranged_Without_Remeasuring()
+			var grid = new Grid();
+			var viewbox = new Viewbox
+			{
+				HorizontalAlignment = HorizontalAlignment.Center,
+				VerticalAlignment = VerticalAlignment.Center,
+				Width = 200,
+				Height = 200,
+				Child = textBox
+			};
+			await UITestHelper.Load(viewbox);
+			await WindowHelper.WaitForIdle();
+
+			var injector = InputInjector.TryCreate() ?? throw new InvalidOperationException("Failed to init the InputInjector");
+			// Double tap the TextBox with finger to ensure we have "touch" based carets
+			using var finger = injector.GetFinger();
+			var textBoxBounds = textBox.GetAbsoluteBoundsRect();
+			var center = textBoxBounds.GetCenter();
+			finger.Press(center);
+			finger.Release();
+			await WindowHelper.WaitFor(() => textBox.FocusState == FocusState.Pointer);
+			textBox.SelectAll();
+
+			// Everything should be selected
+			Assert.AreEqual(0, textBox.SelectionStart);
+			Assert.AreEqual(textBox.Text.Length, textBox.SelectionLength);
+
+			// Wait for the caret popups to appear
+			await WindowHelper.WaitFor(() => VisualTreeHelper.GetOpenPopupsForXamlRoot(WindowHelper.XamlRoot).Where(p => p.Child.FindFirstChild<Microsoft.UI.Xaml.Shapes.Ellipse>() is not null).Any());
+
+			// Get the caret popups
+			var caretPopups = VisualTreeHelper.GetOpenPopupsForXamlRoot(WindowHelper.XamlRoot).Where(p => p.Child.FindFirstChild<Microsoft.UI.Xaml.Shapes.Ellipse>() is not null).ToList();
+			// We should have two caret popups (start and end)
+			Assert.AreEqual(2, caretPopups.Count);
+
+			// Validate the Ellipses of the carets are intersecting the bottom border of the TextBox
+			var textBoxTransform = textBox.TransformToVisual(null);
+			var bottomLeft = textBoxTransform.TransformPoint(new Point(0, textBox.ActualHeight));
+			var bottomRight = textBoxTransform.TransformPoint(new Point(textBox.ActualWidth, textBox.ActualHeight));
+
+			foreach (var popup in caretPopups)
+			{
+				var ellipse = popup.Child.FindFirstChild<Microsoft.UI.Xaml.Shapes.Ellipse>()!;
+				Assert.IsNotNull(ellipse);
+
+				// center in local space
+				var localCenter = new Point(ellipse.Width / 2, ellipse.Height / 2);
+
+				// pick a boundary point on the right side in local space
+				var localBoundary = new Point(ellipse.Width, ellipse.Height / 2);
+
+				// transform both to visual space
+				var ellipseTransform = ellipse.TransformToVisual(null);
+				var ellipseCenter = ellipseTransform.TransformPoint(localCenter);
+				var boundary = ellipseTransform.TransformPoint(localBoundary);
+
+				// actual radius after rotation/scale/etc.
+				double radius = Math.Sqrt(Math.Pow(boundary.X - center.X, 2) + Math.Pow(boundary.Y - center.Y, 2));
+				// Check that the line from bottomLeft to bottomRight intersects the ellipse using DistancePointToSegment
+				var distance = DistancePointToSegment(ellipseCenter, bottomLeft, bottomRight);
+				Assert.IsTrue(distance < radius, "Caret ellipse should intersect the bottom border of the TextBox");
```

Comment:
@MartinZikmund Could we do a screenshot test as well? Maybe with disabling the caret blinking?

Reply:
As the `TextBox` is rotated and scaled, I am not sure what would be the screenshot validation to run there - the ellipse is overlapping the bottom border, but I don't see how to express this as screenshot assert in this case